### ISSUE TITLE
PHP Driver not playing well with Curl

### ISSUE THREAD

**Issue by grepsr** [mongodb/mongo-php-driver#162]
This is a very weird issue. For some reason, very randomly, if you do a Curl request to a new website after initialising the Mongo driver, the eventual "upsert" fails with no proper debug message.

The weird part is, it does not happen with all domains, such as http://yahoo.com/ works. Probably its different for different HTTP response types?

I checked for potential variable name conflicts, but I see none. Can anyone else duplicate this?

Mongo server, mongo client libraries - all latest versions; PHP 5.6

Code below:
```
$manager = new MongoDB\Driver\Manager("mongodb://192.168.99.100/");
$collection = new MongoDB\Collection($manager, "test_database." . 'test_collection');

$indexes = [
    ['key' => ['name' => 1, 'age' => 1], 'unique' => true, 'sparse' => true],
    ['key' => ['timestamp' => 1]]
];
$collection->createIndexes($indexes);
$doc = array('name' => 'john', 'age'=> 20);

// a random curl request
$objCurl = curl_init();
curl_setopt($objCurl, CURLOPT_VERBOSE,1);

// very random behaviour, some website triggers the mongo error below, some do not!!!?!?!!!
curl_setopt($objCurl, CURLOPT_URL, 'http://www.whatever.com');
$result = curl_exec($objCurl);

// mongo upsert
$cond = array('name' => 'john', 'age' => 20);
$ret = $collection->updateOne($cond,
    array('$set' => $doc, '$currentDate' => array('timestamp' => true)),
    array('upsert' => true));
```

Output below:
```
php test.php
* Rebuilt URL to: http://www.whatever.com/
* Hostname was NOT found in DNS cache
*   Trying 198.57.151.250...
* Connected to www.whatever.com (198.57.151.250) port 80 (#0)
> GET / HTTP/1.1
Host: www.whatever.com
Accept: */*

< HTTP/1.1 301 Moved Permanently
* Server nginx/1.8.0 is not blacklisted
< Server: nginx/1.8.0
< Date: Sat, 12 Dec 2015 06:14:11 GMT
< Content-Type: text/html; charset=UTF-8
< Content-Length: 0
< Connection: keep-alive
< Location: http://whatever.com/
<
* Connection #0 to host www.whatever.com left intact
[12-Dec-2015 06:14:12] ERR (3):
#0 /home/grepsr/vortex-backend/library/MongoDB/Operation/Update.php(107): MongoDB\Driver\Server->executeBulkWrite('test_database.t...', Object(MongoDB\Driver\BulkWrite), Object(MongoDB\Driver\WriteConcern))
#1 /home/grepsr/vortex-backend/library/MongoDB/Operation/UpdateOne.php(66): MongoDB\Operation\Update->execute(Object(MongoDB\Driver\Server))
#2 /home/grepsr/vortex-backend/library/MongoDB/Collection.php(564): MongoDB\Operation\UpdateOne->execute(Object(MongoDB\Driver\Server))
#3 /home/grepsr/vortex-backend/test.php(36): MongoDB\Collection->updateOne(Array, Array, Array)
```

**Comment by grepsr** [comment]
Okay so this was due to the socket check bug #144 

The curl was actually delaying the execution, and that caused #144 to occur.

This has been fixed in 1.0.1! 

Closing issue.

**Comment by bjori** [comment]
Thanks for the update @grepsr !
